File: archive/_dev_drafts/irpp_algo_2026.js
function calculerIRPP2026_Corrected(revenuBrutMensuel, typeRevenu = 'salarie', chefDeFamille = false, nbEnfants = 0, nbInfirmes = 0, nbParents = 0, nbEtudiants = 0) {
    console.log(`\n--- SIMULATION FISCALE 2026 ---`);
    console.log(`Données : Brut=${revenuBrutMensuel} DT | Type=${typeRevenu} | Chef=${chefDeFamille} | Enfants=${nbEnfants}`);

    // 1. Constantes & Taux
    const CNSS_TAUX = 0.0918;
    const FRAIS_PRO_TAUX = 0.10;
    const FRAIS_PRO_PLAFOND = 2000.0;
    const ABATTEMENT_RETRAITE = 0.25;
    const DEDUC_CHEF = 300.0;
    const DEDUC_ENFANT = 100.0;
    const DEDUC_INFIRME = 2000.0;
    const DEDUC_PARENT = 400.0;
    const DEDUC_ETUDIANT = 1000.0; // Ajout étudiant LF 2026
    const CSS_TAUX = 0.005;

    // Barème Officiel JORT 148 (8 tranches)
    const BAREME = [
        { min: 0, max: 5000, rate: 0 },
        { min: 5000, max: 10000, rate: 0.15 },
        { min: 10000, max: 20000, rate: 0.25 },
        { min: 20000, max: 30000, rate: 0.30 },
        { min: 30000, max: 40000, rate: 0.33 },
        { min: 40000, max: 50000, rate: 0.36 },
        { min: 50000, max: 70000, rate: 0.38 },
        { min: 70000, max: Infinity, rate: 0.40 }
    ];

    // 2. Calculs
    const brutAnnuel = revenuBrutMensuel * 12;
    let cnss = 0;
    let abattement = 0;

    if (typeRevenu === 'salarie') {
        cnss = brutAnnuel * CNSS_TAUX;
        const brutApresCnss = brutAnnuel - cnss;
        abattement = Math.min(brutApresCnss * FRAIS_PRO_TAUX, FRAIS_PRO_PLAFOND);
    } else {
        // Pension de retraite : pas de CNSS, abattement 25%
        abattement = brutAnnuel * ABATTEMENT_RETRAITE;
    }

    let deductions = 0;
    if (chefDeFamille) deductions += DEDUC_CHEF;
    deductions += Math.min(nbEnfants, 4) * DEDUC_ENFANT;
    deductions += nbInfirmes * DEDUC_INFIRME;
    deductions += nbParents * DEDUC_PARENT;
    deductions += nbEtudiants * DEDUC_ETUDIANT;

    const assiette = Math.max(0, (brutAnnuel - cnss - abattement) - deductions);

    // 3. IRPP
    let irpp = 0;
    BAREME.forEach(tranche => {
        if (assiette > tranche.min) {
            const baseTranche = Math.min(assiette, tranche.max) - tranche.min;
            irpp += baseTranche * tranche.rate;
        }
    });

    // 4. CSS (Règle : Pas d'IRPP = Pas de CSS)
    let css = 0;
    if (irpp > 0) {
        css = assiette * CSS_TAUX;
    }
    // 5. Résultats
    const totalImpot = irpp + css;
    const netAnnuel = brutAnnuel - cnss - totalImpot;
    const netMensuel = netAnnuel / 12;

    console.log("\n--- RÉSULTATS CORRIGÉS ---");
    console.log(`Type de Revenu     : ${typeRevenu.toUpperCase()}`);
    console.log(`Brut Mensuel       : ${revenuBrutMensuel.toFixed(3)} DT`);
    console.log(`Assiette Imposable : ${assiette.toFixed(3)} DT`);
    console.log(`IRPP Dû            : ${irpp.toFixed(3)} DT`);
    console.log(`CSS Dû             : ${css.toFixed(3)} DT`);
    console.log(`NET MENSUEL        : ${netMensuel.toFixed(3)} DT`);
    console.log(`--------------------------------`);

    return { irpp, css, netMensuel };
}

// CAS DE TEST REEL : Pension 400 DT, Chef Famille, 1 Enfant Infirme
// Résultat attendu : IRPP=0, CSS=0, Net=400
calculerIRPP2026_Corrected(400, 'retraite', true, 0, 1, 0);

// Test Salaire Élevé pour vérifier le barème
calculerIRPP2026_Corrected(5000, 'salarie', true, 2, 0, 0);

File: fiscal/archive/_dev_drafts/irpp_algo_2026.py
import math

def calculer_irpp_2026(salaire_brut_mensuel, chef_de_famille, nb_enfants, nb_etudiants):
    """
    Calcul rigoureux de l'IRPP et CSS selon la Loi de Finances Tunisie 2026.
    """
    
    # 1. Constantes
    CNSS_TAUX = 0.0918
    FRAIS_PRO_TAUX = 0.10
    FRAIS_PRO_PLAFOND = 2000.0
    DEDUC_CHEF_FAMILLE = 300.0
    DEDUC_ENFANT = 100.0
    DEDUC_ETUDIANT = 1000.0
    CSS_TAUX = 0.005 # 0.5%
    
    LIMIT_ENFANTS_STD = 4
    
    # Barème 2026
    BAREME = [
        (0, 5000, 0.0),
        (5000, 10000, 0.15),
        (10000, 20000, 0.25),
        (20000, 30000, 0.30),
        (30000, 40000, 0.33),
        (40000, 50000, 0.36),
        (50000, 70000, 0.38),
        (70000, float('inf'), 0.40)
    ]

    # 2. Calculs Initiaux
    salaire_brut_annuel = salaire_brut_mensuel * 12
    
    # 3. CNSS
    cnss_annuelle = salaire_brut_annuel * CNSS_TAUX
    salaire_brut_apres_cnss = salaire_brut_annuel - cnss_annuelle
    
    # 4. Frais Professionnels
    frais_pros = min(salaire_brut_apres_cnss * FRAIS_PRO_TAUX, FRAIS_PRO_PLAFOND)
    
    net_imposable_base = salaire_brut_apres_cnss - frais_pros
    
    # 5. Déductions Familiales
    deductions = 0.0
    
    if chef_de_famille:
        deductions += DEDUC_CHEF_FAMILLE
        
    # Enfants (Max 4 pour les 100 DT)
    nb_enfants_valids = min(nb_enfants, LIMIT_ENFANTS_STD)
    deductions += (nb_enfants_valids * DEDUC_ENFANT)
    
    # Etudiants (Specific deduction)
    deductions += (nb_etudiants * DEDUC_ETUDIANT)
    
    # 6. Assiette Imposable
    assiette = max(0, net_imposable_base - deductions)
    
    # 7. Calcul IRPP par tranches
    irpp_total = 0.0
    details_tranches = []
    
    print(f"\n--- DÉTAIL CALCUL TRANCHES (Assiette : {assiette:.3f} DT) ---")
    
    for (lower, upper, rate) in BAREME:
        if assiette > lower:
            # La base taxable dans cette tranche est la partie de l'assiette qui tombe dedans
            # C'est le min(assiette, upper) - lower
            base_tranche = min(assiette, upper) - lower
            
            tax_tranche = base_tranche * rate
            irpp_total += tax_tranche
            
            if base_tranche > 0 or rate == 0:
                 print(f"Tranche {lower}-{upper} ({rate*100}%): {base_tranche:.3f} DT -> {tax_tranche:.3f} DT")
    
    # 8. CSS
    css_total = assiette * CSS_TAUX
    
    # 9. Totaux et Net
    total_impot = irpp_total + css_total
    
    net_annuel = salaire_brut_annuel - cnss_annuelle - total_impot
    net_mensuel = net_annuel / 12
    
    return {
        "Brut Annuel": salaire_brut_annuel,
        "CNSS": cnss_annuelle,
        "Frais Pros": frais_pros,
        "Déductions Famille": deductions,
        "Assiette Imposable": assiette,
        "IRPP": irpp_total,
        "CSS": css_total,
        "Total Impôt": total_impot,
        "Net Mensuel": net_mensuel
    }

# --- EXEMPLE D'EXECUTION (TEST) ---
# Cas : Salarié, 2400 DT Brut/mois, Chef de famille, 2 enfants dont 1 étudiant
res = calculer_irpp_2026(2400, True, 2, 1)

print("\n--- RÉSULTATS FINAUX ---")
for k, v in res.items():
    print(f"{k}: {v:.3f} DT")
